x-pack/plugins/infra/public/alerting/metric_threshold/components/expression_chart.tsx

```diff
@@ -0,0 +1,284 @@
+  h: i18n.translate('xpack.infra.metrics.alerts.timeLabels.hours', { defaultMessage: 'hours' }),
+  d: i18n.translate('xpack.infra.metrics.alerts.timeLabels.days', { defaultMessage: 'days' }),
+};
+
+export const ExpressionChart: React.FC<Props> = ({ expression, loading, data, id }) => {
+  const metric = {
+    field: expression.metric,
+    aggregation: expression.aggType as MetricsExplorerAggregation,
+    color: MetricsExplorerColor.color0,
+  };
+  const isDarkMode = false; // TODO: Need to figure out a way to get this value through alerts service or context.
+  const dateFormatter = useMemo(() => {
+    const firstSeries = data ? first(data.series) : null;
+    return firstSeries && firstSeries.rows.length > 0
+      ? niceTimeFormatter([first(firstSeries.rows).timestamp, last(firstSeries.rows).timestamp])
+      : (value: number) => `${value}`;
+  }, [data]);
+
+  const yAxisFormater = useCallback(createFormatterForMetric(metric), [expression]);
+
+  if (loading) {
+    return (
+      <EmptyContainer>
+        <EuiText color="subdued">Loading chart...</EuiText>
+      </EmptyContainer>
+    );
+  }
+  if (!data) {
+    return (
+      <EmptyContainer>
+        <EuiText color="subdued">Oops, no chart data available</EuiText>
+      </EmptyContainer>
+    );
+  }
+
+  const thresholds = expression.threshold.slice().sort();
+
+  // Creating a custom series where the ID is changed to 0
+  // so that we can get a proper domian
+  const firstSeries = first(data.series);
+  if (!firstSeries) {
+    return (
+      <EmptyContainer>
+        <EuiText color="subdued">Oops, no chart data available</EuiText>
+      </EmptyContainer>
+    );
+  }
+
+  const series = {
+    ...firstSeries,
+    rows: firstSeries.rows.map(row => {
+      const newRow: MetricsExplorerRow = {
+        timestamp: row.timestamp,
+        [`metric_0`]: row[`metric_${id}`] || null,
+      };
+      thresholds.forEach((thresholdValue, index) => {
+        newRow[`metric_threshold_${index}`] = thresholdValue;
+      });
+      return newRow;
+    }),
+  };
+
+  const firstTimestamp = first(firstSeries.rows).timestamp;
+  const lastTimestamp = last(firstSeries.rows).timestamp;
+  const dataDomain = calculateDomain(series, [metric], false);
+  const domain = {
+    max: Math.max(dataDomain.max, last(thresholds) || dataDomain.max) * 1.1, // add 10% headroom.
+    min: Math.min(dataDomain.min, first(thresholds) || dataDomain.min),
+  };
+
+  if (domain.min === first(expression.threshold)) {
+    domain.min = domain.min * 0.9;
+  }
+
+  const isAbove = [Comparator.GT, Comparator.GT_OR_EQ].includes(expression.comparator);
+  const opacity = 0.1;
+  const timeLabel = TIME_LABELS[expression.timeUnit];
+
+  return (
+    <>
+      <ChartContainer>
+        <Chart>
+          <MetricExplorerSeriesChart
+            type={MetricsExplorerChartType.area}
+            metric={metric}
+            id="0"
```

Comment:
Is  `id="0"` valid?

Reply:
Yes, normally in metrics explorer this would be part of a `map` and that value would be set the the same `index` of the metric (series) that's being rendered. Since we are "faking" the expected structure on line 100, its should always be `0`.